### ISSUE TITLE
Handling empty schemaPattern in querydsl-maven-plugin

### ISSUE THREAD

**Issue by metacubed** [querydsl/querydsl#2282]
In the querydsl-maven-plugin. when the `schemaPattern` parameter is specified as:
```xml
<schemaPattern></schemaPattern>
```
it is converted to `null` instead of an empty string. This is a well-known Maven limitation.

However, these two values have different meanings when the [`DatabaseMetadata`](https://docs.oracle.com/javase/8/docs/api/java/sql/DatabaseMetaData.html#getTables-java.lang.String-java.lang.String-java.lang.String-java.lang.String:A-) class processes them:
```
"" retrieves those without a schema
<code>null</code> means that the schema name should not be used to narrow the search
```
There was a similar issue raised earlier, for the `namePrefix` parameter. This was handled by using the special string `BLANK` to represent an empty string value.

> **PR #1811 - Fix empty value handling**

Can we implement similar logic for the `schemaPattern` parameter (including multi-value handing)?

**Comment by Shredder121** [comment]
That sounds good to me!

**Comment by metacubed** [comment]
in progress

**Comment by metacubed** [comment]
Fixed in #2286
